src/genh5_datatype.h

```diff
@@ -378,9 +382,13 @@ GENH5_DECLARE_DATATYPE(char, DataType::Char);
 GENH5_DECLARE_DATATYPE(char*, DataType::VarString);
 GENH5_DECLARE_DATATYPE(char const*, DataType::VarString);
 
+GENH5_DECLARE_DATATYPE(int8_t, DataType::Int8);
+GENH5_DECLARE_DATATYPE(int16_t, DataType::Int16);
 GENH5_DECLARE_DATATYPE(int, DataType::Int);
 GENH5_DECLARE_DATATYPE(long int, DataType::Long);
 GENH5_DECLARE_DATATYPE(long long int, DataType::LLong);
+GENH5_DECLARE_DATATYPE(uint8_t, DataType::UInt8);
+GENH5_DECLARE_DATATYPE(uint16_t, DataType::Int16);
 GENH5_DECLARE_DATATYPE(unsigned int, DataType::UInt);
```

Comment:
We may have to add a compile-time flag if `int` is (at least) 32bit? However, then again on the platforms we wantto support this should be the case.

Reply:
We should switch to int32_t and int64_t in general I think instead of long, long long etc